src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -265,13 +266,16 @@ void IExtensionConfigProvider.Initialize(ExtensionConfigContext context)
             // Throw if any of the configured options are invalid
             this.Options.Validate(this.nameResolver, this.TraceHelper);
 
-            // For 202 support
-            if (this.Options.NotificationUrl == null)
-            {
 #pragma warning disable CS0618 // Type or member is obsolete
-                this.Options.NotificationUrl = context.GetWebhookHandler();
-#pragma warning restore CS0618 // Type or member is obsolete
+
+            // Invoke webhook handler to make functions runtime register extension endpoints.
+            context.GetWebhookHandler();
+
+            if (this.Options.TestWebhookUri == null)
+            {
+                this.HttpApiHandler.RegisterWebhookProvider(() => context.GetWebhookHandler());

```

Comment:
This line does quite a lot of heavy lifting: by registering the provider like this, we make sure `context.WebhookHandler` gets called every time we need to URI. Since this line is so crucial, can we comment it with a note on what this achieves?